cdragontoolbox/patcher.py

```diff
@@ -163,20 +163,24 @@ def _parse_table(parser, entry_parser):
     @staticmethod
     def _parse_bundle(parser):
         """Parse a bundle entry"""
-        _, n, bundle_id = parser.unpack('<llQ')
-        # skip remaining header part, if any
-        parser.skip(n - 12)
 
-        bundle = PatcherBundle(bundle_id)
-        n, = parser.unpack('<l')
-        for _ in range(n):
-            pos = parser.tell()
-            offset, = parser.unpack('<l')
-            parser.seek(pos + offset)
-            parser.skip(4)  # skip offset table offset
-            compressed_size, uncompressed_size, chunk_id = parser.unpack('<LLQ')
+        def parse_chunklist(parser):
+            fields = PatcherManifest._parse_field_table(parser, (
+                ('chunk_id', '<Q'),
+                ('compressed_size', '<L'),
+                ('uncompressed_size', '<L')
+            ))
+            yield fields['chunk_id'], fields['compressed_size'], fields['uncompressed_size']
+
+        fields = PatcherManifest._parse_field_table(parser, (
+            ('bundle_id', '<Q'),
+            ('chunks_offset', 'offset')
+        ))
+
+        bundle = PatcherBundle(fields['bundle_id'])
+        parser.seek(fields['chunks_offset'])
+        for (chunk_id, compressed_size, uncompressed_size), in PatcherManifest._parse_table(parser, parse_chunklist):
```

Comment:
That doesn't actually work because the `yield fields['chunk_id'], fields['compressed_size'], fields['uncompressed_size']` expression actually returns a generator which returns a single tuple. This is the most concise syntax I've found that implicitly evaluates that generator without needing extra steps.

Reply:
Hum... indeed.

But `parse_chunklist` does not need to be generator. (Other calls to `_parse_table()` don't use one.) If it was a regular function you could just use a `for` withing the extra comma.